WordPress/src/main/java/org/wordpress/android/ui/posts/editor/ImageEditorInitializer.kt

```diff
@@ -58,17 +60,22 @@ class ImageEditorInitializer {
         }
 
         private fun <T : Any> onResourceReady(model: Any?, listener: ImageEditor.RequestListener<T>, resource: T) =
-            if (model != null && (model is String || model is Uri)) {
-                listener.onResourceReady(resource, model.toString())
-            } else {
-                throw(IllegalArgumentException(IMAGE_STRING_URL_MSG))
-            }
+                listener.onResourceReady(resource, getResourcePath(model))
 
         private fun <T : Any> onLoadFailed(model: Any?, listener: ImageEditor.RequestListener<T>, e: Exception?) =
+                listener.onLoadFailed(e, getResourcePath(model))
+
+        private fun getResourcePath(model: Any?): String {
```

Comment:
Could you please elaborate why we need to get the actual path here? I think it's considered a bad practice - see https://commonsware.com/blog/2016/03/15/how-consume-content-uri.html for more details. I tried to remove the code and the app still seemed to work. However, I haven't done thorough tests.

Reply:
I was trying to be safe and consistent with the local file uris usage and found actual path being used in other parts of the app like here :

https://github.com/wordpress-mobile/WordPress-Android/blob/36a05e94de9f0e2632000cb26ec7f69d131435b2/WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java#L372-L379

I tested and haven't yet found a case when app doesn't work without it. Also we don't have such a conversion when we pass the uris to the preview fragment here: 

https://github.com/wordpress-mobile/WordPress-Android/blob/80c39533a3fb4eb08db616927583dc7cb1cab562/WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.java#L688-L691

So reverting it here: 1a8b0cb

Reply:
I see, thanks for the clarification!